openff/toolkit/utils/toolkits.py

```diff
@@ -168,6 +169,10 @@ class AntechamberNotFoundError(MessageException):
 # UTILITY FUNCTIONS
 # =============================================================================================
 
+def mol_to_ctab_and_aro_key(self, molecule, aromaticity_model=DEFAULT_AROMATICITY_MODEL):
+    return f"{molecule.ordered_connection_table_hash()}-{aromaticity_model}"
```

Comment:
From @connordavel: Could add a kwarg here for `safe_caching` which would default to `True`. If it's true, this would use `to_dict` as a slow, but safe molecule hash. Otherwise it would use this janky connection table hash for speed (but it loses metadata, so most people won't want it)